### DISCUSSION TITLE
How winget select the installer?

### DISCUSSION THREAD

**Discussion by Guillem2607** [microsoft/winget-cli#2785] [Q&A] [answered]
Habitually winget have a AppXInstaller.yaml that contains a string that is headed as InstallerURL, as example [7-zip](https://raw.githubusercontent.com/microsoft/winget-pkgs/master/manifests/7/7zip/7zip/22.01/7zip.7zip.installer.yaml). So it have, Installers>Achitectures(x64/x86/arm64) and then there is .EXE and .MSI installers.

![wingetinstaller](https://user-images.githubusercontent.com/121028740/208630647-c29c909f-4907-4dc3-abf4-150c507d4908.PNG)

So I'm trying to understand how winget chose the installer for a local project in a server, with an script pull the package and overwrite the last package if there is one before and then this package that is saved in a local repository will be pulled by local machines of the net that are being configured, so I'm reducing time by reducing the time of pulling from [Winget Repository](https://github.com/microsoft/winget-pkgs/tree/master/manifests).

I hope someone could response me :-)

**Comment by Trenly** [comment, marked as answer]
Winget uses several parameters to choose the installer. It looks to make sure that the MinimumOSVersion is met, and then tries to satisfy any requirements set in the user settings or command line parameters. If no installers meet the requirements, winget will exit. If multiple installers remain, winget will try and match the architecture which is currently running on the machine. From there it will use a heuristic match to select the best installer - if there is a currently installed version of the package it will use the same install technology.
